modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java

```diff
@@ -572,36 +593,149 @@ private long _getRootObjectEntryAccountEntryId(
 			accountEntryRestrictedObjectField.getName());
 	}
 
+	private Date _getStartDate() {
+		return Date.from(
+			LocalDate.now(
+			).minusDays(
+				Objects.equals(_objectConfiguration.timeScale(), "days") ?
+					_objectConfiguration.duration() :
+						_objectConfiguration.duration() * 7
+			).atStartOfDay(
+				ZoneId.systemDefault()
+			).toInstant());
+	}
+
+	private void _sendNotificationToAdminUsers(
+			long adminUserId, String portletId,
+			ObjectDefinition objectDefinition)
+		throws Throwable {
+
+		TransactionInvokerUtil.invoke(
```

Comment:
We must send all notifications in the same transaction, so move the administrator user IDs loop to the body of the TransactionInvokerUtil

Reply:
Did you have a chance to change this?